src/unstructured.rs

```diff
@@ -316,30 +316,48 @@ impl<'a> Unstructured<'a> {
         // When there is only one possible choice, don't waste any entropy from
         // the underlying data.
         if start == end {
-            return Ok((*start, 0));
+            return Ok((start, 0));
         }
 
-        let range: T::Widest = end.as_widest() - start.as_widest();
-        let mut result = T::Widest::ZERO;
-        let mut offset: usize = 0;
+        let delta = end - start;
```

Comment:
we might need to modify the Int trait to have an associated sized_type int type that we have conversions to/from

Reply:
Not sure I follow what you are suggesting?